pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -342,6 +345,137 @@ func setupKernelTunables(option KernelTunableBehavior) error {
 	return utilerrors.NewAggregate(errList)
 }
 
+func setQOSCgroupMemoryLimit(cgroupManager CgroupManager, qosClass v1.PodQOSClass, limit int64) error {
```

Comment:
the only issue i see there is that in the case that the Update() fails for setting the memory limit, we want to do extra stuff i.e. find the usage and set the limit at the usage.